src/pybind/mgr/dashboard_v2/controllers/rbd_mirroring.py

```diff
@@ -0,0 +1,314 @@
+    def default(self):
+        status, content_data = self._get_content_data()
+        return {'status': status, 'content_data': content_data}
+
+    @ViewCache()
+    def _get_daemons_and_pools(self):  # pylint: disable=R0915
+        def get_daemons():
+            daemons = []
+            for hostname, server in CephService.get_service_map('rbd-mirror').items():
+                for service in server['services']:
+                    id = service['id']  # pylint: disable=W0622
+                    metadata = service['metadata']
+                    status = service['status']
+
+                    try:
+                        status = json.loads(status['json'])
+                    except ValueError:
+                        status = {}
+
+                    instance_id = metadata['instance_id']
+                    if id == instance_id:
+                        # new version that supports per-cluster leader elections
+                        id = metadata['id']
+
+                    # extract per-daemon service data and health
+                    daemon = {
+                        'id': id,
+                        'instance_id': instance_id,
+                        'version': metadata['ceph_version'],
+                        'server_hostname': hostname,
+                        'service': service,
+                        'server': server,
+                        'metadata': metadata,
+                        'status': status
+                    }
+                    daemon = dict(daemon, **get_daemon_health(daemon))
+                    daemons.append(daemon)
+
+            return sorted(daemons, key=lambda k: k['instance_id'])
+
+        def get_daemon_health(daemon):
+            health = {
+                'health_color': 'info',
+                'health': 'Unknown'
+            }
+            for _, pool_data in daemon['status'].items():
+                if (health['health'] != 'error' and
+                        [k for k, v in pool_data.get('callouts', {}).items()
+                         if v['level'] == 'error']):
+                    health = {
+                        'health_color': 'error',
+                        'health': 'Error'
+                    }
+                elif (health['health'] != 'error' and
+                      [k for k, v in pool_data.get('callouts', {}).items()
+                       if v['level'] == 'warning']):
+                    health = {
+                        'health_color': 'warning',
+                        'health': 'Warning'
+                    }
+                elif health['health_color'] == 'info':
+                    health = {
+                        'health_color': 'success',
+                        'health': 'OK'
+                    }
+            return health
+
+        def get_pools(daemons):  # pylint: disable=R0912, R0915
+            pool_names = [pool['pool_name'] for pool in CephService.get_pool_list('rbd')]
+
+            if pool_names is None:
```

Comment:
AFAICT pool_names will never equal to None

Reply:
Yes, I agree.